File: src/kyu_8/alternating_case/cpp/solution.cpp
std::string to_alternating_case(const std::string& str)
{
    std::string result;
    result.reserve(str.size()); // Предварительное выделение памяти под строку `result` того же размера, что и str
    std::transform(str.begin(), str.end(), std::back_inserter(result),
        [](unsigned char c) { return std::tolower(c) == c ? std::toupper(c) : std::tolower(c); });
    return result;
}

File: src/kyu_8/alternating_case/csharp/solution.cs
using System;

namespace Extensions
{
  public static class StringExt
  {
    public static string ToAlternatingCase (this string s)
    {
        string result = "";
        foreach (char c in s.ToCharArray())
        {
            if(Char.IsUpper(c))
            {
                result += Char.ToLower(c);
            } else {
                result += Char.ToUpper(c);
            }
        }
        return result;
    }
  }
}